packages/xo-server/src/index.mjs

```diff
@@ -752,6 +769,10 @@ export default async function main(args) {
         if (req.secure) {
           return next()
         }
+        // don't redirect let's encrypt challenge to https
+        if (req.url.startsWith('/.well')) {
+          return next()
+        }
```

Comment:
To make sure Express does not interfere, you can update this line: https://github.com/vatesfr/xen-orchestra/blob/af87d6a0eaaf71349af4b7af67ba58518a9c7cfc/packages/xo-server/src/index.mjs#L764